### DISCUSSION TITLE
Best practise for createMany large record sizes

### DISCUSSION THREAD

**Discussion by daniel-glynn** [prisma/prisma#13660] [Q&A]
Hey everyone, have googled a bit and found some half-answers but was wondering did anyone have some good input to the below :)

So using Prisma I'm going to start batching large amounts of records into the DB using createMany, these amounts will vary anywhere from 100-20,000. I was wondering is it more performant to use createMany even when dealing with the high point of 20,000 records, or batch them into smaller chunks of 1-2k instead? Keeping in mind if even one fails I don't want any of them to be inserted, so if they were batched into chunks of 1-2k they'd all have to be done as part of one transaction which might be less performant then one createMany operation?

Really appreciate any advice :)

Regards,
Daniel

**Comment by austincrim** [comment]
Hey @daniel-glynn 👋,

I think the easiest way to find out would be to test each method and record the results.

Is this something you are planning on doing regularly? May I ask what the use case is?

**Reply by daniel-glynn** [reply]
Hey @austincrim , thanks for the response!:)

Yeah I figured we'd have to do some experimenting as part of the implementation, was just throwing it out there incase anyone had some good experience.

Our usecase will be a user notifications crud server which will be consuming events sent out from another server, each event having an attached list of users which we want to fan out and insert that many notifications for. So one event with 1000 users attached equals 1000 inserts into the DB.

Depending on how many users need to be notified per event, this could range anywhere from 10s to 1000s. 

Thanks for the reply, much appreciated.

**Reply by austincrim** [reply]
Very cool! It would be great if you could update this issue with the findings from your experiments. I am curious to know the results!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
